boa/src/builtins/regexp/mod.rs

```diff
@@ -714,4 +720,67 @@ impl RegExp {
 
         Ok(result)
     }
+
+    /// `RegExp.prototype[ @@search ]( string )`
+    ///
+    /// This method executes a search for a match between a this regular expression and a string.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-regexp.prototype-@@search
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@search
+    pub(crate) fn search(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        // 1. Let rx be the this value.
+        // 2. If Type(rx) is not Object, throw a TypeError exception.
+        if let Some(object) = this.as_object() {
```

Comment:
If you want to avoid indenting the rest of the function you can use `match` to either unwrap or early return, something like this:

```rust
let object = match this.as_object() {
  Some(object) => object,
  None => return context.throw_type_error("RegExp.prototype.search method called on incompatible value"),
}
```

ref: https://doc.rust-lang.org/rust-by-example/error/result/early_returns.html

Reply:
You can also use `if !this.is_object()` and then `expect()` on the `as_object()`.